src/app/containers/reference-form-dialog/index.tsx

```diff
@@ -0,0 +1,121 @@
+import React, { useCallback, useState, ChangeEvent } from 'react';
+import { useDispatch } from 'react-redux';
+
+import { createBlankReference, Reference, ReferencePerson, ReferenceType } from 'app/models/reference';
+import { Select } from 'app/components/select';
+import { useReferenceFormStyles } from 'app/containers/reference-form-dialog/styles';
+import { ActionButton } from 'app/components/action-button';
+import * as manuscriptEditorActions from 'app/actions/manuscript-editor.actions';
+import * as manuscriptActions from 'app/actions/manuscript.actions';
+import { renderConfirmDialog } from 'app/components/prompt-dialog';
+import { ReferenceAuthorsList } from 'app/containers/reference-form-dialog/reference-authors-list';
+
+interface ReferenceFormDialogProps {
+  reference?: Reference;
+}
+
+export const ReferenceFormDialog: React.FC<ReferenceFormDialogProps> = ({ reference }) => {
+  const classes = useReferenceFormStyles();
+  const [isConfirmShown, setConfirmSnow] = useState<boolean>(false);
+  const isNewReference = !reference;
+  const [userReference, setReference] = useState<Reference>(reference || createBlankReference());
+  const dispatch = useDispatch();
+
+  const handleReferenceTypeChange = useCallback(
+    (event: ChangeEvent<{ name: string; value: ReferenceType }>) => {
+      setReference({
+        ...userReference,
+        type: event.target['value']
+      });
+    },
+    [userReference]
+  );
+
+  const closeDialog = useCallback(() => {
+    dispatch(manuscriptEditorActions.hideModalDialog());
+  }, [dispatch]);
+
+  const handleAuthorsListChange = useCallback(
+    (refAuthors: ReferencePerson[]) => {
+      setReference({
+        ...userReference,
+        authors: refAuthors
+      });
+    },
+    [userReference, setReference]
+  );
+
+  const handleDone = useCallback(() => {
+    userReference.authors = userReference.authors.filter(
+      (author) => author['firstName'] || author['lastName'] || author['groupName']
+    );
+    if (isNewReference) {
+      dispatch(manuscriptActions.addReferenceAction(userReference));
+    } else {
+      dispatch(manuscriptActions.updateReferenceAction(userReference));
+    }
+    closeDialog();
+  }, [userReference, closeDialog, dispatch, isNewReference]);
+
+  const handleReject = useCallback(() => {
+    setConfirmSnow(false);
```

Comment:
Typo? I guess it's no big deal but would be nice if it was corrected to be I assume 'setConfirmShow'